pkg/k8s/client.go

```diff
@@ -88,3 +91,38 @@ func (c *KubeClient) DestinationRuleCRDExists(ctx context.Context) (bool, error)
 	}
 	return true, nil
 }
+
+func (c *KubeClient) DeletePVCsWithLabel(ctx context.Context, labelSelector string,
+	mustHaveNamePrefix, namespace string) error {
+	// create a new labels.Selector object for the label selector
+	selector, err := labels.Parse(labelSelector)
+	if err != nil {
+		return err
+	}
+
+	pvcList := &apiv1.PersistentVolumeClaimList{}
+	if err = c.client.List(ctx, pvcList, &client.ListOptions{
+		Namespace:     namespace,
+		LabelSelector: selector,
+	}); err != nil {
+		return err
+	}
+
+	// if there are no PVCs in the list, do nothing.
+	if len(pvcList.Items) == 0 {
+		return nil
+	}
+
+	// delete each PVC in the list.
+	for i := range pvcList.Items {
+		pvc := pvcList.Items[i]
+		// pvc.Name string starts with "eventing-"
+		if strings.HasPrefix(pvc.Name, mustHaveNamePrefix) {
+			err = c.client.Delete(ctx, &pvc)
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
(nit) Shall we add a log message saying that we are ignoring PVC deletion because of name prefix not found?